hoomd/state.py

```diff
@@ -38,33 +39,166 @@ def _create_domain_decomposition(device, box):
 class State:
     """The state of a `hoomd.Simulation` object.
 
-    Provides access (read/write) to a `hoomd.Simulation` object's particle,
-    bond, angle, etc. data. Data access is facilitated through two complementary
-    APIs: *global* and *local* snapshots (note that global does not refer to
-    variable scope here). See `State.snapshot`, `State.cpu_local_snapshot`, and
-    `State.gpu_local_snapshot` for information about these data access patterns.
-    In addition, many commonly used smaller quantities such as the number of
-    particles in a simulation are available directly through `State` object
-    properties.
+    Note:
+        This object cannot be directly instantiated. Use
+        `hoomd.Simulation.create_state_from_gsd` and
+        `hoomd.Simulation.create_state_from_snapshot` to instantiate a `State`
+        object as part of a simulation.
+
+    .. rubric:: Overview
+
+    `State` stores the data that describes the thermodynamic microstate of a
+    `hoomd.Simulation` object. This data consists of the box, particles, bonds,
+    angles, dihedrals, impropers, special pairs, and constraints.
+
+    .. rubric:: Box
+
+    The simulation `box` describes the area in space that contains the particles
+    and the periodic boundary conditions to apply. See `Box` for a full
+    description.
+
+    .. rubric:: Particles
+
+    The state contains `N_particles` particles. Each particle has a position,
+    orientation, type id, body, mass, moment of inertia, charge, diameter,
+    velocity, angular momentum, image, and tag:
+
+    - :math:`\\vec{r}`: position :math:`[\\mathrm{length}]` - x,y,z cartesian
+      coordinates defining the position of the particle in the box
+    - :math:`\\mathbf{q}`: orientation :math:`[\\mathrm{dimensionless}]` - (s,
+      :math:`\\vec{a}`), unit quaternion defining the rotation from the
+      particle's local reference frame to the box reference frame.
+    - :math:`t_\\mathrm{id}`: type id :math:`[\\mathrm{dimensionless}]` -
+      integer in the range ``[0,len(particle_types)``) that identifies the
+      particle's type. `particle_types` maps type ids to names with:
+      ``name = particle_types[t_id]``.
+    - :math:`b_\\mathrm{id}`: body id :math:`[\\mathrm{dimensionless}]` -
+      integer that identifies the particle's body. A value of ``-1`` indicates
```

Comment:
```suggestion
      integer that identifies the particle's rigid body. A value of ``-1`` indicates
```
I think using rigid as an adjective the first time helps users understand what is meant by body.